lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1141,13 +1141,42 @@ public function get_custom_css() {
 		// Add the global styles root CSS.
 		$stylesheet = _wp_array_get( $this->theme_json, array( 'styles', 'css' ), '' );
 
+		// Add the global styles element CSS.
+		if ( isset( $this->theme_json['styles']['elements'] ) ) {
+			foreach ( $this->theme_json['styles']['elements'] as $element => $node ) {
+				$custom_element_css = _wp_array_get( $this->theme_json, array( 'styles', 'elements', $element, 'css' ) );
+				if ( $custom_element_css ) {
+					$selector    = static::ELEMENTS[ $element ];
+					$stylesheet .= $this->process_blocks_custom_css( $custom_element_css, $selector );
+				}
+			}
+		}
+
 		// Add the global styles block CSS.
 		if ( isset( $this->theme_json['styles']['blocks'] ) ) {
 			foreach ( $this->theme_json['styles']['blocks'] as $name => $node ) {
 				$custom_block_css = _wp_array_get( $this->theme_json, array( 'styles', 'blocks', $name, 'css' ) );
 				if ( $custom_block_css ) {
-					$selector    = static::$blocks_metadata[ $name ]['selector'];
-					$stylesheet .= $this->process_blocks_custom_css( $custom_block_css, $selector );
+					// Only use the class for the default style variation if the block has style variations.
+					if ( isset( static::$blocks_metadata[ $name ]['styleVariations'] ) ) {
+						// Add the block selector and the .is-style-default style variation class.
+						$selector    = static::$blocks_metadata[ $name ]['selector'] . ', ' . static::$blocks_metadata[ $name ]['styleVariations']['default'];
```

Comment:
The default variation isn't always named "default". This won't work for the Button block, for example, and we also don't have any way of enforcing the naming for third party blocks. The only way to make sure a style variation is default is by checking for the "isDefault" property.